### PR TITLE
cmds: implement `mem` read/write tool

### PR DESCRIPTION
**Pull request by gerard5**
The `mem` command provides the ability to test physical memory adresss space, `mem` tool reads and writes memory cells. When it comes to physical memory space inspection (read), it differs from the `dump` command in that the reading is performed by specifying the size of the word, not bytes, and values are read as integers of specified width machine order. Additionally memory alignment may be specified, it checks addresses with `plo` memory map definitions (to disable use `-F` switch).

### Some usage examples

read 32bit word from `0x7c00` address
```bash
(plo)% mem -r -- 0x7c00
```
*note `--` may be skipped, is optional to separate options/switches from the physical address

write `0xdeadc0de` value at `0x7c00` address
```bash
(plo)% mem -w 0xdeadc0de -- 0x7c00
```

write 256 `ff's`  starting from `0x20020000` address (memset)
```bash
(plo)% mem -c 256 -s 1 -w 0xff -- 0x20020000
```

write 1KB `ff` block starting from `0x20020000` address (memset), default size=4, alignment=4
```bash
(plo)% mem -c 256 -w 0xffffffff -- 0x20020000
```

write 2 bytes (0x55) in the middle of 32bit word at`0x20020000`
```bash
(plo)% mem -c 2 -a 1 -s 1 -w 0x55 -- 0x20020001
```

Read a 64bit value from`0x404a0402` (address will be aligned down to 0x404a0400).
```bash
(plo)% mem -a 8 -s 8 -r -- 0x404a0402
```

If address is `not writable`, `not readable` use `-F` switch to disable memory map checks (and likely expect an exception for an invalid address).
```bash
(plo)% mem -F -r -- 0x87654321
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: imxrt-1176-nil, ia32-generic, armv7a9-zynq7000-qemu

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: cmds/mem.c [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		u16 val16;
+		u8 val8;
+	} u = { .val64 = 0uLL };
+
+	if (sizeof(addr_t) <= 8uLL) {
+		lib_printf("read 0x%08x = ", addr);
+	}
+	else {
+		lib_printf("read 0x%016llx = ", (u64)addr);
+	}
```
Comment (agkaminski) [inline cmds/mem.c]:
```suggestion
	lib_printf("read 0x%p = ", (void *)addr);
```

File: cmds/mem.c [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+	if ((forceValid != 0) || (region_validator(addr, size, mAttrRead) != 0)) {
+		switch (size) {
+			case sizeof(u.val8):
+				hal_cpuDataMemoryBarrier();
+				u.val8 = *(u8 *)addr;
+				lib_printf("0x%02x\n", u.val8);
```
Comment (agkaminski) [inline cmds/mem.c:73]: Intermediate variable isn't really needed, but it's a matter of taste, feel free to ignore this remark

```suggestion
				lib_printf("0x%02x\n", *(u8 *)addr);
```

File: cmds/mem.c (hunk 1 above)
Comment (gerard5) [reply cmds/mem.c]: I've been here already :), `0x%p` for `0` address returns weird things.

2023-08-07-093722_1889x86_scrot

Comment (agkaminski) [reply cmds/mem.c]: Then maybe NULL check only? Accessing NULL is rare use-case anyway

**Review by agkaminski — DISMISSED** [review]

File: cmds/mem.c (hunk 1 above)
Comment (gerard5) [reply cmds/mem.c]: Ok, changed to `%p` version.

**Review by agkaminski — DISMISSED** [review]

**Review by agkaminski — APPROVED** [review]

**Review by FPiorski — APPROVED** [review]

**Pull request merged**